tasks/spec_runner.rake

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'rspec/core'
+require 'test_queue'
+require 'test_queue/runner/rspec'
+
+# A TestQueue framework that is explicitly given RSpec arguments instead of
+# implicitly reading ARGV.
+class ParallelSpecFramework < TestQueue::TestFramework::RSpec
+  def initialize(rspec_args)
+    @rspec_args = rspec_args
+  end
+
+  def all_suite_files
+    options = ::RSpec::Core::ConfigurationOptions.new(@rspec_args)
+    options.parse_options if options.respond_to?(:parse_options)
+    options.configure(::RSpec.configuration)
+
+    ::RSpec.configuration.files_to_run.uniq
+  end
+end
+
+# Heavily inspired by TestQueue::Runner::RSpec
+class ParallelSpecRunner < TestQueue::Runner
+  def initialize(rspec_args)
+    super(ParallelSpecFramework.new(rspec_args))
+  end
+
+  def run_worker(iterator)
+    rspec = ::RSpec::Core::QueueRunner.new
+    rspec.run_each(iterator).to_i
+  end
+
+  def summarize_worker(worker)
+    worker.summary = worker.lines.grep(/\A\d+ examples?, /).first
+    worker.failure_output = worker.output[/^Failures:\n\n(.*)\n^Finished/m, 1]
+  end
+end
+
+def with_encoding(encoding)
+  previous_encoding = Encoding.default_external
+  Encoding.default_external = encoding
+  yield
+ensure
+  Encoding.default_external = previous_encoding
+end
+
+def run_specs(encoding: 'UTF-8')
```

Comment:
`#with_encoding` and `#run_specs` should probably not be defined in the global namespace. Should I add a module for wrapping these two together with the Framework and the Runner above?